### PR TITLE
Polish new extension menu view UI

### PR DESCRIPTION
**Pull request by sangwoo108**
Part of https://github.com/brave/brave-browser/issues/37583

image

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
* Enable flag brave://flags/#extensions-menu-access-control
* See new extensions menu

### CODE REVIEW

**Comment by simonhong** [comment]
Hmm, it seems this PR is incomplete.
I'm seeing when control pin state from three dot menu. and don't see toggle button?
image

**Comment by sangwoo108** [comment]
> I'm seeing when control pin state from three dot menu. 

I'm not sure what you're meaning

> and don't see toggle button?

Toggle button is hidden by upstream when current site is not where extension can't be work on. e.g. chrome://scheme. This is upstream's behavior not ours.

> They are pinned but unpinned icon is shown.

I guess that's pinned icon. We don't show unpinned icon.

**Comment by simonhong** [comment]
> > I'm seeing when control pin state from three dot menu.
> 
> I'm not sure what you're meaning

Demo by contro pin state with three dot menu. The issue is upstream and our icons are visible both.
https://github.com/brave/brave-core/assets/6786187/79edb4d5-5e42-423b-8484-044ef3cd5931


> 
> > and don't see toggle button?
> 
> Toggle button is hidden by upstream when current site is not where extension can't be work on. e.g. chrome://scheme. This is upstream's behavior not ours.

Ah, right. if site doesn't request any permission, toggle is not shown.
> 
> > They are pinned but unpinned icon is shown.
> 
> I guess that's pinned icon. We don't show unpinned icon.

**Comment by sangwoo108** [comment]
> Demo by contro pin state with three dot menu. The issue is upstream and our icons are visible both.

Aha! Thanks, I didn't know that

**Comment by sangwoo108** [comment]
> Demo by contro pin state with three dot menu. The issue is upstream and our icons are visible both.

Fixed!

File: browser/ui/views/extensions/brave_extensions_menu_main_page_view.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (simonhong) [inline browser/ui/views/extensions/brave_extensions_menu_main_page_view.h:1]:
```suggestion
```

**Review by simonhong — APPROVED** [review]
++ 👍🏼

**Pull request merged**